tests/syscalls/test_mpi_rng.c

```diff
@@ -0,0 +1,112 @@
+#include <setjmp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+// must come after setjmp.h
+#include <cmocka.h>
+
+#include <openssl/bn.h>
+#include <openssl/rand.h>
+
+#include "bolos/cxlib.h"
+#include "utils.h"
+
+unsigned long sys_cx_rng(uint8_t *buffer, unsigned int length);
+
+void test_mpi_rng(void **state __attribute__((unused)))
+{
+  cx_mpi_t *r, *n;
+  int ret, one, two;
+  cx_err_t error;
+  uint32_t i, j, rnd, value;
+  uint8_t buffer[64];
+
+  // Those tests will check that cx_mpi_rng is working as expected:
+  // cx_mpi_rng(r,n) => Generate a random value r in the range 0 < r < n.
+
+  r = BN_new();
+  assert_non_null(r);
+  n = BN_new();
+  assert_non_null(n);
+
+  // Check that cx_mpi_rng returns CX_INVALID_PARAMETER if n=0:
+  ret = BN_set_word(n, 0);
+  assert_int_equal(ret, 1);
+  error = cx_mpi_rng(r, n);
+  assert_int_equal(error, CX_INVALID_PARAMETER);
+
+  // Check that cx_mpi_rng returns CX_INVALID_PARAMETER if n=1:
+  ret = BN_set_word(n, 1);
+  assert_int_equal(ret, 1);
+  error = cx_mpi_rng(r, n);
+  assert_int_equal(error, CX_INVALID_PARAMETER);
+
+  // Check that cx_mpi_rng returns CX_OK and r=1 if n=2:
+  ret = BN_set_word(n, 2);
+  assert_int_equal(ret, 1);
+  for (i = 0; i < 256; i++) {
+    error = cx_mpi_rng(r, n);
+    assert_int_equal(error, CX_OK);
+    value = BN_get_word(r);
```

Comment:
`BN_is_one`?

Reply:
I don't understand what you are suggesting, here: could you please provide the full code of the suggestion?

Reply:
I wasn't looking in the right place, sorry.
I understood what you suggested and I will do it ;-)